zookeeper-server/src/main/java/org/apache/zookeeper/common/StandardTypeFileKeyStoreLoader.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zookeeper.common;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+
+/**
+ * Base class for instances of {@link KeyStoreLoader} which load the key/trust
+ * stores from files on a filesystem using standard {@link KeyStore} types like
+ * JKS or PKCS12.
+ */
+abstract class StandardTypeFileKeyStoreLoader extends FileKeyStoreLoader {
+    private static final Logger LOG = LoggerFactory.getLogger(StandardTypeFileKeyStoreLoader.class);
+
+    private static final char[] EMPTY_CHAR_ARRAY = new char[0];
+
+    StandardTypeFileKeyStoreLoader(String keyStorePath,
+                                   String trustStorePath,
+                                   String keyStorePassword,
+                                   String trustStorePassword) {
+        super(keyStorePath, trustStorePath, keyStorePassword, trustStorePassword);
+    }
+
+    @Override
+    public KeyStore loadKeyStore() throws IOException, GeneralSecurityException {
+        KeyStore ks = keyStoreInstance();
+        InputStream inputStream = null;
```

Comment:
We can use try-with-resources now.
I know this is only a refactoring but we can take this opportunity to clean up code

Reply:
The propagation of `IOException`s on `close()` is prevented (only logging is done), in contrast to other `IOException`s. I guess we need to keep this behaviour, right?
If so, try-with-resources doesn't appear beneficial to me. For instance, this might be the code:
```java
try (InputStream inputStream = new FileInputStream(new File(trustStorePath))) {
    ts.load(inputStream, passwordStringToCharArray(trustStorePassword));
    return ts;
} catch (IOException e) {
    // some cumbersome logic to differentiate IOExceptions
    boolean isExceptionOnClose = ...;
    if (isExceptionOnClose) {
        LOG.info("Failed to close key store input stream", e);
    } else {
        throw e;
    }
}
```
Or I'm missing something?

Reply:
@ivanyu I see your point and I thought about it before adding this comment, thank you for pointing it out explicitly.
An exception on close() is really unexpected and it is a symptom of a weird status of the system  (and we can end with open FD leaks...)

So I don't see much value in keeping the method exiting normally when there is something bad going on.

As a counterpart the try-with-resources is a cleaner syntax.

It is just nitpicking, it is not a blocker for this change to be merged.

How does it sound to you ?

Reply:
I totally share this idea about making the methods fail explicitly if closing of . So I've changed to try-with-resources.